### ISSUE TITLE
Make Console.Out redirect optional

### ISSUE THREAD

**Issue by topalach** [SimonCropp/XunitLogger#44]
By default, `Console.Out` is redirected to `TestWriter` on module init. 

#### Problem

When the test code is also used in a console application (for example for debugging purposes), `Console.WriteLine` etc. is not working.

#### Suggested solution

Make the `Console.Out` redirect optional or expose a method for restoring the default `Console.Out`

#### Workaround

Leverage the module load order to restore the original `Console.Out` value:

1. Store the original `Console.Out` value before `XunitLogger` is loaded
2. Make `XunitLogger` load
3. Redirect `Console.Out` back to the original `Console.Out` value

An example of such workaround can be found here: https://github.com/ravendb/ravendb/pull/9814

**Comment by SimonCropp** [comment]
at what point would we 

> Redirect Console.Out back to the original Console.Out value

**Comment by topalach** [comment]
> at what point would we
> 
> > Redirect Console.Out back to the original Console.Out value

This is the aforementioned (straightforward but not too elegant) workaround:
https://github.com/ravendb/ravendb/pull/9814/commits/2b69c44e7e292ef47406800cf379d137042719f0#diff-7e7e113e2865aec092158f57de23c857R38

**Comment by SimonCropp** [comment]
can u illustrate that in a PR?

**Comment by ppekrol** [comment]
Sometimes it is more convenient to use console app, especially when you want to do more complex stuff like run few tests in a parallel or in a loop. Running it through a test project is not always that easy. We have been doing that for years and found this approach to be the most straightforward.

**Comment by SimonCropp** [comment]
can you share some code that is easier in a console compared to a unit test?

**Comment by ppekrol** [comment]
@SimonCropp we found a more suitable solution, thanks for your time

**Comment by SimonCropp** [comment]
can u share what that solution is?

**Comment by SimonCropp** [comment]
@ppekrol @topalach bump?

**Comment by ppekrol** [comment]
Hi @SimonCropp 

We figured out that the main issue was with us using `dotnet-xunit` with verbose and Jenkins. Both of them do not work well when console is redirected (empty log in Jenkins). As the `dotnet-xunit` is depracated, we didn't want to introduce changes to your library for all of the users. We just forked and added switch to turn off redirects. After we will move to `dotnet test` it should not be a problem.
